compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -532,9 +534,12 @@ private void addToCompiledSymbols(Set<Symbol> compiledSymbols, Scope.ScopeEntry
                 }
             }
 
-            compiledSymbols.add(compiledSymbol);
+            if (!(symbol.getKind() == SymbolKind.PACKAGE
+                    && !((BPackageSymbol) symbol).compUnit.getValue().equals(compUnitName))) {
```

Comment:
Since this condition is somewhat hard to understand, shall we move this to a a private util method? WDYT?